ec2/spark_ec2.py

```diff
@@ -872,9 +890,16 @@ def deploy_files(conn, root_dir, opts, master_nodes, slave_nodes, modules):
     if "." in opts.spark_version:
         # Pre-built Spark deploy
         spark_v = get_validate_spark_version(opts.spark_version, opts.spark_git_repo)
+        tachyon_v = get_tachyon_version(spark_v)
     else:
         # Spark-only custom deploy
         spark_v = "%s|%s" % (opts.spark_git_repo, opts.spark_version)
+        if opts.tachyon_version is None:
```

Comment:
I see your point, but as @nchammas mentioned only some fraction of Spark usage scenarios involve using Tachyon. If one just wants to say run the `master` branch to do some spark-perf benchmarks expecting the user to set a Tachyon version seems like a unnecessary burden. 

If we don't want to use a default tachyon version I'd vote for just not starting tachyon in cases where a spark git hash is used.  (i.e. remove tachyon from the list of modules and print a info message)